deployments/controller/ka-updater-kured.yaml

```diff
@@ -0,0 +1,182 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: kubearmor-kured
+  namespace: kubearmor
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: kubearmor-kured
+  namespace: kubearmor
+rules:
+- apiGroups:
+  - apps
+  resourceNames:
+  - kubearmor-kured
+  resources:
+  - daemonsets
+  verbs:
+  - update
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kubearmor-kured
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - nodes
+  verbs:
+  - get
+  - patch
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  verbs:
+  - list
+  - delete
+  - get
+- apiGroups:
+  - apps
+  resources:
+  - daemonsets
+  verbs:
+  - get
+- apiGroups:
+  - ""
+  resources:
+  - pods/eviction
+  verbs:
+  - create
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: kubearmor-kured
+  namespace: kubearmor
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: kubearmor-kured
+subjects:
+- kind: ServiceAccount
+  name: kubearmor-kured
+  namespace: kubearmor
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: kubearmor-kured
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: kubearmor-kured
+subjects:
+- kind: ServiceAccount
+  name: kubearmor-kured
+  namespace: kubearmor
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: kubearmor-kured
+  namespace: kubearmor
+spec:
+  selector:
+    matchLabels:
+      name: kured
+  template:
+    metadata:
+      labels:
+        name: kured
+    spec:
+      containers:
+      - args:
+        - |
+          grep "bpf" /rootfs/sys/kernel/security/lsm >/dev/null
+          [[ $? -eq 0 ]] && echo "sysfs already has BPF enabled" && rm /sentinel/reboot-required && sleep infinity
+          grep "GRUB_CMDLINE_LINUX.*bpf" /rootfs/etc/default/grub >/dev/null
+          [[ $? -eq 0 ]] && echo "grub already has BPF enabled" && sleep infinity
```

Comment:
We need to update this logic. There's a lot of other `bpf` related options that can be specified in boot parameters related to bpf. Azure Mariner Linux does mention it and the script assumes that bpflsm is enabled when it's not.